src/app/integrations/bamboo-hr/bamboo-hr.component.ts

```diff
@@ -163,6 +164,22 @@ export class BambooHrComponent implements OnInit {
     });
   }
 
+  tokenExpiredBambooHr(): void {
+    this.isLoading = true;
+    this.bambooHrService.disconnectBambooHr().subscribe(() => {
+      this.displayToastMessage(ToastSeverity.ERROR, 'Session expired on BambooHR! Please connect again to continue.');
+      this.isBambooConnected = false;
+      this.isLoading = false;
+    });
+  }
```

Comment:
We shouldn't disconnect BambooHR here. Disconnect should be called only when you click the 'Disconnect' button in the UI. We want to show a to-do item in the admin dashboard when a token expires - and calling `disconnectBambooHr()` prevents that. Instead, we should directly do:

```this.isBambooConnected = false;```

without calling `disconnectBambooHr()`.
```suggestion
  tokenExpiredBambooHr(): void {
    this.displayToastMessage(ToastSeverity.ERROR, 'Session expired on BambooHR! Please connect again to continue.');
    this.isBambooConnected = false;
  }
```